### ISSUE TITLE
Unable to launch gvim from vscode snap

### ISSUE THREAD

**Issue by jonsmithers** [microsoft/vscode#72322]
- VSCode Version: 1.33.1, 51b0b28134d51361cf996d2f0a1c698247aeabd8, x64
- OS Version: Ubuntu 18.10 and Fedora 29

When installed as a snap package, vscode is unable to properly launch gvim. This prevents [a certain plugin ](https://marketplace.visualstudio.com/items?itemName=jonsmithers.open-in-vim) from functioning properly (jonsmithers/vscode-open-in-vim/issues/4). 

Steps to Reproduce:

1. Install vscode as a snap package (https://snapcraft.io/code)
2. Open terminal (`Ctrl`-`` ` ``)
3. Run `gvim`

   You will encounter the following error in the terminal:

   ```
   Vim: Caught deadly signal SEGV
   Vim: Finished.
   E852: The child process failed to start the GUI
   ```

Does this issue occur when all extensions are disabled?: Yes

**Comment by khoitd1997** [comment]
I just want to add that after it reports the error above, it will enter vim in the cli:

![Like this](https://media.giphy.com/media/cP5SyksCTCrLm165uW/giphy.gif)

**Comment by khoitd1997** [comment]
gvim spawns three processes, here is the stack trace in case you need it.
```
  Stack trace of thread 9796:
#0  0x00007f46ebe301bb kill (libc.so.6)
#1  0x000055d6ca53d1b7 may_core_dump (gvim)
#2  0x000055d6ca64b8be getout (gvim)
#3  0x00007f46ebfd0e70 __restore_rt (libpthread.so.0)
#4  0x00007f46d534ed41 n/a (libpixbufloader-xpm.so)
#5  0x00007f46d534f211 n/a (libpixbufloader-xpm.so)
#6  0x00007f46ec6dee1d gdk_pixbuf_new_from_xpm_data (libgdk_pixbuf-2.0.so.0)
#7  0x000055d6ca5fa2e3 mainwin_realize (gvim)
#8  0x00007f46ec4b0742 g_closure_invoke (libgobject-2.0.so.0)
#9  0x00007f46ec4c44b4 signal_emit_unlocked_R (libgobject-2.0.so.0)
#10 0x00007f46ec4cd34d g_signal_emit_valist (libgobject-2.0.so.0)
#11 0x00007f46ec4cd973 g_signal_emit (libgobject-2.0.so.0)
#12 0x00007f46ecd1a4ce gtk_widget_realize (libgtk-3.so.0)
#13 0x00007f46ecd27dc6 gtk_window_show (libgtk-3.so.0)
#14 0x00007f46ec4b0742 g_closure_invoke (libgobject-2.0.so.0)
#15 0x00007f46ec4c45f9 signal_emit_unlocked_R (libgobject-2.0.so.0)
#16 0x00007f46ec4cd34d g_signal_emit_valist (libgobject-2.0.so.0)
#17 0x00007f46ec4cd973 g_signal_emit (libgobject-2.0.so.0)
#18 0x00007f46ecd1451e gtk_widget_show (libgtk-3.so.0)
#19 0x000055d6ca5ff466 gui_mch_open (gvim)
#20 0x000055d6ca5f3381 gui_init (gvim)
#21 0x000055d6ca5c3063 set_termname (gvim)
#22 0x000055d6ca5f4cab gui_attempt_start (gvim)
#23 0x000055d6ca5f4f86 gui_do_fork (gvim)
#24 0x000055d6ca64cfe6 vim_main2 (gvim)
#25 0x000055d6ca42d497 main (gvim)
#26 0x00007f46ebe1bf33 __libc_start_main (libc.so.6)
#27 0x000055d6ca42f06e _start (gvim)

Stack trace of thread 9797:
#0  0x00007f46ebee82b7 __GI___poll (libc.so.6)
#1  0x00007f46ec3c92de g_main_context_poll (libglib-2.0.so.0)
#2  0x00007f46ec3c9413 g_main_context_iteration (libglib-2.0.so.0)
#3  0x00007f46ec3c9461 glib_worker_main (libglib-2.0.so.0)
#4  0x00007f46ec3f2752 g_thread_proxy (libglib-2.0.so.0)
#5  0x00007f46ebfc65a2 start_thread (libpthread.so.0)
#6  0x00007f46ebef3023 __clone (libc.so.6)

Stack trace of thread 9798:
#0  0x00007f46ebee82b7 __GI___poll (libc.so.6)
#1  0x00007f46ec3c92de g_main_context_poll (libglib-2.0.so.0)
#2  0x00007f46ec3c96b3 g_main_loop_run (libglib-2.0.so.0)
#3  0x00007f46ec60fafa gdbus_shared_thread_func (libgio-2.0.so.0)
#4  0x00007f46ec3f2752 g_thread_proxy (libglib-2.0.so.0)
#5  0x00007f46ebfc65a2 start_thread (libpthread.so.0)
#6  0x00007f46ebef3023 __clone (libc.so.6)
```

**Comment by Tyriar** [comment]
FYI @flexiondotorg

**Comment by Kedstar99** [comment]
When I ran 'snap run --shell code', I entered a shell environment that is equivalent to the running snap.
In this context, I was able to run gvim and it would spawn without errors. 
There were also no denials logged in dmesg, nor in /var/log/syslog.

This is based on the instructions posted here for security confinement.
https://docs.ubuntu.com/core/en/guides/intro/security

This doesn't particularly seem like a snap confinement issue otherwise im fairly certain I would be recording the same behaviour in the raw shell environment.
